client/service/src/lib.rs

```diff
@@ -181,136 +181,158 @@ pub struct ServiceComponents<TBl: BlockT, TBackend: Backend<TBl>, TSc, TExPool,
+							sc_rpc::system::PeerInfo {
+								peer_id: peer_id.to_base58(),
+								roles: format!("{:?}", p.roles),
+								protocol_version: p.protocol_version,
+								best_hash: p.best_hash,
+								best_number: p.best_number,
+							}
+						).collect());
+					}
+					sc_rpc::system::Request::NetworkState(sender) => {
+						if let Some(network_state) = serde_json::to_value(&network.network_state()).ok() {
+							let _ = sender.send(network_state);
 						}
-						Err(e) => sender.send(Err(sc_rpc::system::error::Error::MalformattedPeerArg(
-							e.to_string(),
-						))),
-					};
-				}
-				sc_rpc::system::Request::NodeRoles(sender) => {
-					use sc_rpc::system::NodeRole;
+					}
+					sc_rpc::system::Request::NetworkAddReservedPeer(peer_addr, sender) => {
+						let x = network.add_reserved_peer(peer_addr)
+							.map_err(sc_rpc::system::error::Error::MalformattedPeerArg);
+						let _ = sender.send(x);
+					}
+					sc_rpc::system::Request::NetworkRemoveReservedPeer(peer_id, sender) => {
+						let _ = match peer_id.parse::<PeerId>() {
+							Ok(peer_id) => {
+								network.remove_reserved_peer(peer_id);
+								sender.send(Ok(()))
+							}
+							Err(e) => sender.send(Err(sc_rpc::system::error::Error::MalformattedPeerArg(
+								e.to_string(),
+							))),
+						};
+					}
+					sc_rpc::system::Request::NodeRoles(sender) => {
+						use sc_rpc::system::NodeRole;
 
-					let node_role = match role {
-						Role::Authority { .. } => NodeRole::Authority,
-						Role::Light => NodeRole::LightClient,
-						Role::Full => NodeRole::Full,
-						Role::Sentry { .. } => NodeRole::Sentry,
-					};
+						let node_role = match role {
+							Role::Authority { .. } => NodeRole::Authority,
+							Role::Light => NodeRole::LightClient,
+							Role::Full => NodeRole::Full,
+							Role::Sentry { .. } => NodeRole::Sentry,
+						};
 
-					let _ = sender.send(vec![node_role]);
+						let _ = sender.send(vec![node_role]);
+					}
 				}
-			};
-		}
+			}
 
-		// Interval report for the external API.
-		status_sinks.lock().poll(cx, || {
-			let status = NetworkStatus {
-				sync_state: network.sync_state(),
-				best_seen_block: network.best_seen_block(),
-				num_sync_peers: network.num_sync_peers(),
-				num_connected_peers: network.num_connected_peers(),
-				num_active_peers: network.num_active_peers(),
-				average_download_per_sec: network.average_download_per_sec(),
-				average_upload_per_sec: network.average_upload_per_sec(),
-			};
-			let state = network.network_state();
-			(status, state)
-		});
-
-		// Main network polling.
-		if let Poll::Ready(Ok(())) = Pin::new(&mut network).poll(cx).map_err(|err| {
-			warn!(target: "service", "Error in network: {:?}", err);
-		}) {
-			return Poll::Ready(());
+			// The network worker has done something. Nothing special to do, but could be
+			// used in the future to perform actions in response of things that happened on
+			// the network.
+			_ = (&mut network).fuse() => {}
```

Comment:
We were previously logging the error returned by the `NetworkWorker` future:

```rust
impl<B: BlockT + 'static, H: ExHashT> Future for NetworkWorker<B, H> {
	type Output = Result<(), io::Error>;
```

Is this not necessary anymore?

Reply:
Error are never generated. I believe that the output is a `Result` because in `futures 0.1` you were forced to return a `Result`, and we never changed that.

Reply:
Even better. Let me check. If so I will create a follow up pull request.

Reply:
See https://github.com/paritytech/substrate/pull/6552.